packages/transformers/graphql/src/GraphQLTransformer.js

```diff
@@ -1,58 +1,40 @@
 // @flow
 import {Transformer} from '@parcel/plugin';
 
-const IMPORT_RE = /^# *import +['"](.*)['"] *;? *$/;
-const NEWLINE = /\r\n|[\n\r\u2028\u2029]/;
+import {parse, print, Source} from 'graphql/language';
+import {stripIgnoredCharacters} from 'graphql/utilities';
+import {processDocumentImports} from 'graphql-import-macro';
 
 export default new Transformer({
-  async transform({asset, options, resolve}) {
-    // Peer dependency of graphql-tag
-    await options.packageManager.resolve('graphql', asset.filePath);
-
-    let gql = await options.packageManager.require(
-      'graphql-tag',
-      asset.filePath,
-      {autoinstall: options.autoinstall},
-    );
-
-    let gqlMap = new Map();
-    const traverseImports = (name: string, assetCode: string) => {
-      gqlMap.set(name, assetCode);
-
-      return Promise.all(
-        assetCode
-          .split(NEWLINE)
-          .map(line => line.match(IMPORT_RE))
-          .filter(match => !!match)
-          // $FlowFixMe
-          .map(async ([, importName]) => {
-            let resolved = await resolve(name, importName);
-
-            if (gqlMap.has(resolved)) {
-              return;
-            }
-
-            let code = await options.inputFS.readFile(resolved, 'utf8');
-            asset.addIncludedFile({
-              filePath: resolved,
-            });
-
-            await traverseImports(resolved, code);
-          }),
-      );
-    };
+  getConfig({options}) {
+    return Promise.resolve({
+      stripIgnoredCharacters: options.mode === 'production',
+    });
+  },
+
+  async transform({asset, options, resolve, config}) {
+    const document = parse(new Source(await asset.getCode(), asset.filePath));
+
+    const expandedDocument = await processDocumentImports(document, loadImport);
 
-    await traverseImports(asset.filePath, await asset.getCode());
+    const generated =
+      config && config.stripIgnoredCharacters
```

Comment:
You can just access `options` directly here, using `getConfig` provides no benefit in this case.

As I already said in the linked issue, `asset.env.minify` fits better here than `mode`, this way specifying `--no-minify` on the CLI might help debugging.